### PR TITLE
Use drydock at system planet with highest happiness

### LINKED ISSUE #1122: Orbital Drydock does not use best available dryock in system.
zarquan, posted the following on the forum.  I have not verified it, but it is possible. 

### Description
I've copied zarquan's comment from the [forum](http://freeorion.org/forum/viewtopic.php?f=28&t=10328&sid=08be4798b31e8c09fa04edabb8930cdf) below:

"
Hello. First of all, I really like the changes to the orbital drydocks. It makes conquests harder and gives the defenders an advantage.

Unfortunately, there is a minor bug involving more than one orbital drydock in a system. If there are two orbital drydocks in a system, but the closer planet has no population, the ships still try to repair there, even if the further planet has a fully functional drydock. I don't fully understand the mechanics at this point. I researched Advanced Damage Control, and the ships only seem to be repaired from that. I get the message:
"On Ephyse I: the Orbital Drydock is abandoned, there is no one left to assist in repairs of the ship Deer."

I ran the game a few more turns (avoiding my ships being repaired by Damage Control techs) and I got it to the point where the drydock at Ephyse I was functioning. I sent the ships to be repaired there, and I got this messge:
"On Ephyse II: the Orbital Drydock completed repairs on the Shale"

I don't understand the mechanics enough to attempt to diagnose anything.

I would attach the save file, but it is too big.

I am running FreeOrion [v0.4.6](url) [build 2016-09-16.49f9123] MSVC 2013 on Windows 10.

"

### Environment

* **FreeOrion Version**: [v0.4.6](url) [build 2016-09-16.49f9123] MSVC 2013 
* **Operating System**: Windows 10.

### PR DESCRIPTION
**Pull request by dbenage-cx**
Limits effect activation to the system planet (with a drydock) that has the highest happiness.

Fixes #1122

### CODE REVIEW

File: default/scripting/buildings/shipyards/ORBITAL_DRYDOCK.focs.txt [hunk 1]
```diff
@@ -35,6 +35,7 @@ BuildingType
                 Source
                 ContainedBy And [
                     Planet
+                    [[PLANET_OWNED_DRYDOCK_HIGHEST_HAPPINESS]]
```
Comment (Vezzra) [inline default/scripting/buildings/shipyards/ORBITAL_DRYDOCK.focs.txt:37]: Correct me if I'm wrong, but isn't the `Planet` condition redundant here, as `PLANET_OWNED_DRYDOCK_HIGHEST_HAPPINESS` already matches only planets...?

Comment (dbenage-cx) [reply default/scripting/buildings/shipyards/ORBITAL_DRYDOCK.focs.txt:37]: TY. Fixed

**Comment by MatGB** [comment]
I've been swamped with Fighter stuff recently and not had time to test this, but it all looks fine from this end.

**Comment by geoffthemedio** [comment]
@MatGB reminder

**Comment by Vezzra** [comment]
@MatGB, do you think you'll be able to get around to test this a bit in the forseeable future? If not, maybe we can just merge this? As you noted, the changes look sound AFAICT.

**Comment by MatGB** [comment]
I forgot to cherry pick it in but it looks fine and simple to me, assuming it runs merge and fix weirdness if we find it.

**Comment by Vezzra** [comment]
Done.

**Pull request merged**